content/en/blog/_posts/2022-12-6-provision-volumes-from-cross-namespace-snapshots-alpha.md

```diff
@@ -0,0 +1,133 @@
+
+
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/deploy/kubernetes/snapshot-controller/rbac-snapshot-controller.yaml
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml
+   ```
+
+2. Deploy External Provisioner controller.
+
+   ```terminal
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-provisioner/master/deploy/kubernetes/deployment.yaml 
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-provisioner/master/deploy/kubernetes/rbac.yaml
+   ```
+
+3. Deploy csi-hostpath driver.
+
+   ```terminal
+   cd /tmp
+   git clone --depth=1 https://github.com/kubernetes-csi/csi-driver-host-path.git
+   cd csi-driver-host-path
+   ./deploy/kubernetes-latest/deploy.sh
+   ```
+
+4. Deploy ReferenceGrants CRD
+  
+   ```terminal
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/main/config/ crd/experimental/gateway.networking.k8s.io_referencegrants.yaml
+   ```
+
+5. Create StorageClass, PVC, and VolumeSnapshot by the examples in the csi-hostpath repo
+
+    ```terminal
+    cd /tmp/csi-driver-host-path
+    kubectl apply -f examples/csi-storageclass.yaml
+    kubectl apply -f examples/csi-pvc.yaml
+    kubectl apply -f examples/csi-snapshot-v1beta1.yaml
+    ```
+
+6. Ceate a ns1 namespace
+
+    ```terminal
+    kubectl create ns ns1
+    ```
+
+7. Create a ReferenceGrant
+
+   ```yaml
+   apiVersion: gateway.networking.k8s.io/v1alpha2
+   kind: ReferenceGrant
+   metadata:
+     name: bar
+     namespace: default
+   spec:
+     from:
+     - group: ""
+       kind: PersistentVolumeClaim
+       namespace: ns1
+     to:
+     - group: snapshot.storage.k8s.io
+       kind: VolumeSnapshot
+       name: new-snapshot-demo
+   ```
+
+8. Create a PersistentVolumeClaim
+
+   ```yaml
+   apiVersion: v1
+   kind: PersistentVolumeClaim
+   metadata:
+     name: foo-pvc
+     namespace: ns1
+   spec:
+     storageClassName: csi-hostpath-sc
+     accessModes:
+     - ReadWriteOnce
+     resources:
+       requests:
+         storage: 1Gi
+     dataSourceRef2:
+       apiGroup: snapshot.storage.k8s.io
+       kind: VolumeSnapshot
+       name: new-snapshot-demo
+       namespace: default
+     volumeMode: Filesystem
+   ```
+
+Note This is only the simplest example.
+
+## How can I learn more?
+
+The enhancement proposal,
+[Provision volumes from cross-namespace snapshots](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/3294-provision-volumes-from-cross-namespace-snapshots), includes lots of detail about the history and technical implementation of this feature.
+
+Please get involved by joining the Kubernetes storage SIG to help us enhance this
```

Comment:
```suggestion
Please get involved by joining the Kubernetes SIG Storage to help us enhance this
```

Reply:
Actually, “the Kubernetes storage SIG” is fine. Expanded out, it'd be “the Kubernetes storage special interest group” which is idiomatic.